### PR TITLE
Add new eyebrow banner component

### PR DESCRIPTION
**Pull request by rezrah**
## Summary

Part of https://github.com/github/primer/issues/2779

Adds new eyebrow banner component.

🔗 [Documentation](https://primer-c8a6ab8d60-26139705.drafts.github.io/components/EyebrowBanner)
🔗 [Storybook](https://primer-c8a6ab8d60-26139705.drafts.github.io/storybook/?path=/story/components-eyebrowbanner-features--variations&globals=colorMode:dark)

Todo in follow up PR:

- [ ] Test coverage
- [ ] Integration tests

## List of notable changes:

- **added** eyebrow banner component
- **added**  new `eyebrow` slot to Hero component, to enable placement of an eyebrow banner (amongst other things)

## What should reviewers focus on?

- Need additional middle step values for gradient colors
- Review of light mode, for which we don't have any prior art

## Steps to test:

1. [Use this storybook link](https://primer-c8a6ab8d60-26139705.drafts.github.io/storybook/?path=/story/components-eyebrowbanner-features--variations&globals=colorMode:dark) and review the individual stories and playground
1. [Use this documentation link](https://primer-c8a6ab8d60-26139705.drafts.github.io/components/EyebrowBanner) to read through public docs

## Contributor checklist:

- [x] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

Playground walkthrough:

https://github.com/primer/brand/assets/13340707/37864018-1df3-4210-9054-4cb97e12a809

Hero placement:

Screenshot 2023-11-02 at 15 05 54

### CODE REVIEW

**Review by auareyou — APPROVED** [review]
Looks good!

**Comment by jesussandreas** [comment]
Looks good @rezrah.

Couple changes suggestions.
- Let's remove the linear gradient we have and substitute it with `#0D1117 at 60% opacity` since it creates legibility issues.
- On mobile, let's make the border radius 16px to avoid the weird shape it's getting.
- Also on Mobile, can we make the font-size `font-size: var(--brand-text-size-100);`? I think 200 it's too big in comparison to the text.

Thatss all.

**Comment by rezrah** [comment]
Thanks @jesussandreas. All feedback addressed... 

https://github.com/primer/brand/assets/13340707/38fd35e5-6158-451b-8f95-3fba067900ba

**Review by jesussandreas — APPROVED** [review]

File: apps/docs/scripts/components-with-animation.js [hunk 1]
```diff
@@ -11,7 +11,6 @@ export const supportedComponents = [
   'Pillar',
   'SectionIntro',
   'Stack',
-  'Pillar',
```
Comment (joseph-lozano) [inline apps/docs/scripts/components-with-animation.js:14]: Was this an intentional deletion?

**Review by joseph-lozano — APPROVED** [review]
There looks like a stray change in apps/docs/scripts/components-with-animation.js, but otherwise looks great!

File: apps/docs/scripts/components-with-animation.js (hunk 1 above)
Comment (rezrah) [reply apps/docs/scripts/components-with-animation.js:14]: Yeah it's a dupe of the one above (line 11).. no idea how this got inserted here, maybe post-merge conflict 🤷‍♂️.  This file itself is generated, and isn't being built deterministically right now.. so you can ignore changes to this file for time being until that's fixed.

**Pull request merged**